### PR TITLE
imlement issue #480 : UDP target host address is cached

### PR DESCRIPTION
**Pull request by lxhoan**
this PR is a simpler solution for #480

### CODE REVIEW

**Comment by lxhoan** [comment]
Hi @fmachado , @majst01 , in case we have dns resolving cached at JVM level, still need this PR to fix issue #480

**Comment by lxhoan** [comment]
Hi @majst01 , @fmachado
What do you think ?
tks

**Comment by majst01** [comment]
Hi @lxhoan 

I like this approach, but we should at least measure what performance impact this gives ?

File: src/main/java/org/influxdb/impl/InfluxDBImpl.java [hunk 1]
```diff
@@ -187,18 +188,20 @@ public InfluxDBImpl(final String url, final String username, final String passwo
     setRetentionPolicy(retentionPolicy);
   }
 
-  private InetAddress parseHostAddress(final String url) {
-      HttpUrl httpUrl = HttpUrl.parse(url);
+  private String parseHost(final String url) {
+    HttpUrl httpUrl = HttpUrl.parse(url);
```
Comment (fmachado) [inline src/main/java/org/influxdb/impl/InfluxDBImpl.java]: Is `HttpUrl` able to parse URIs that do not start with the string `http`? IMO we should treat this parameter as an `URI` and not an `URL`. The main reasons are:
* Users may finally stop using the scheme "http://" to connect via UDP. [RFC 1738 for URL](https://tools.ietf.org/html/rfc1738#section-3) does not allow the scheme "udp://" but the [RFC 3986 for URI](https://tools.ietf.org/html/rfc3986#section-3.1) does.
* If so, this line could be replaced with `URI.create(...)` and later you could use `URI#getHost()` to get only the hostname.

**Comment by lxhoan** [comment]
@fmachado , I have fixed as your comment, tks

**Comment by lxhoan** [comment]
Hi @majst01 , 

>I like this approach, but we should at least measure what performance impact this gives ?

Performance impact should be inconsiderable since DNS resolving is cached at JVM level (as we discussed intensively before), so most of ```InetSocketAddress``` resolving is are just intra-process invocations

**Comment by lxhoan** [comment]
closed (PR #499 as an alternative)

**Pull request merged**